authorization/v2/uniformResourceIdentifier.go

```diff
@@ -52,3 +110,30 @@ func getRequestArguments(ctx *gin.Context) map[string]string {
 
 	return args
 }
+
+func unmarshalArguments(source string) (map[string]string, error) {
+	arguments := strings.Split(source, "&")
+	return unmarshallURIList(arguments)
+}
+
+func unmarshallURIList(uriList []string) (map[string]string, error) {
+	toReturn := map[string]string{}
+	for _, arg := range uriList {
+		split := strings.Split(arg, "=")
+		toReturn[split[0]] = split[1]
+	}
+	return toReturn, nil
+}
+
+func marshallURIMap(uriMap map[string]string) string {
+	var marshalledMap = ""
```

Comment:
`var marshalledMap string` would be cleaner. `""` is the zero value for a string

Reply:
Oops, sorry, I'm still not used to Go, old habits it seems

Reply:
nw